File: singleton/Singleton.java
public class Singleton {
private static uniqueInstance = null ; 

private Singleton(){
System.out.println('Singleton') ; 
} 

public static Singleton getInstance(){
	if(uniqueInstance ===null){
		uniqueInstance = new Singleton();
	}
	
		return uniqueInstance ; 
}

}

File: php/Singleton/Singleton.php
<?php


class Singleton {

 private static $_instance = null ; 


 private function __construct () {
    // self::$_instance=Singleton::getInstance();
     echo  "je suis l'instance num ". uniqid();
 }


 public  static function getInstance(){
        if(self::$_instance == null){
            self::$_instance = new Singleton();
        }
        return self::$_instance;
 }

}